src/components/ProductDetails.js

```diff
@@ -0,0 +1,100 @@
+import React, { useContext } from "react";
+import "../css/ProductDetails.css";
+import "../App.css";
+import { useParams } from "react-router-dom";
+import { Link } from "react-router-dom";
+import { UseProductDetails } from "../Hooks/UseProductDetails";
+import { CartContext } from "../context/cart-context";
+import { WishListContext } from "../context/wishlist-context";
+import { wishlistDataHandler, presentInWishlist } from "../utils/wishlistUtils";
+import { AddToCart } from "../api/CartApi";
+import { presentInCart } from "../utils/cartUtils";
+import {LoaderSpinner} from './LoaderSpinner';
+
+export const ProductDetails = () => {
+  const { id } = useParams();
+  const productDetail = UseProductDetails(id);
+  const { products, cartItem, cartId, dispatch, loader } = useContext(CartContext);
+  const { wishlistId, wishList, WishlistDispatch } = useContext(
```

Comment:
You can put the useContext in a custom hook itself inside the context.js file.  refer this [link](https://dev.to/alexandprivate/use-react-context-as-a-custom-hook-4ioc) to know more about it